BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -43,53 +56,560 @@ public struct AttributedLabel: Element, Hashable {
+                    guard NSIntersectionRange(string.entireRange, link.range).length > 0 else {
+                        return nil
+                    }
+
+                    return LinkAccessibilityElement(
+                        container: self,
+                        label: string.attributedSubstring(from: link.range).string,
+                        link: link
+                    )
+                }
+        }
+
+        func applyLinkColors(activeLinks: [Link] = []) {
+            let mutableString = NSMutableAttributedString(attributedString: attributedText ?? .init(string: ""))
+
+            for link in links {
+                mutableString.addAttributes(
+                    [.foregroundColor: linkColor],
+                    range: link.range
+                )
+            }
+
+            for link in activeLinks {
+                mutableString.addAttributes(
+                    [.foregroundColor: activeLinkColor],
+                    range: link.range
+                )
+            }
+
+            attributedText = mutableString
+        }
+
+        override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+            guard let first = touches.first else { return }
+            let touchedLinks = links(at: first.location(in: self))
+            trackingLinks = touchedLinks
+            applyLinkColors(activeLinks: touchedLinks)
+        }
+
+        override func touchesMoved(_ touches: Set<UITouch>, with event: UIEvent?) {
+            guard let first = touches.first, let trackingLinks = trackingLinks else { return }
+            let touchedLinks = links(at: first.location(in: self))
+            let activeLinks = Set(touchedLinks).intersection(trackingLinks)
+            applyLinkColors(activeLinks: Array(activeLinks))
+        }
+
+        override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent?) {
+            guard let first = touches.first, let trackingLinks = trackingLinks else { return }
+            let touchedLinks = links(at: first.location(in: self))
+            let activeLinks = Set(touchedLinks).intersection(trackingLinks)
+            for link in activeLinks {
+                urlHandler?.open(link: link)
+            }
+
+            self.trackingLinks = nil
+            applyLinkColors()
+        }
+
+        override func touchesCancelled(_ touches: Set<UITouch>, with event: UIEvent?) {
+            trackingLinks = nil
+            applyLinkColors()
+        }
+    }
+
+    private final class LinkAccessibilityElement: UIAccessibilityElement {
+        private let link: AttributedLabel.Link
+        private weak var container: LabelView?
+
+        init(
+            container: LabelView,
+            label: String,
+            link: AttributedLabel.Link
+        ) {
+            self.link = link
+            self.container = container
+            super.init(accessibilityContainer: container)
+            accessibilityLabel = label
```

Comment:
Do we also need to set traits on this? (I don't think we do, but wanted to double check)

Reply:
No, setting them didn't change anything